### ISSUE TITLE
The simplest way to use checkpoint to maximize the GPU memory usage

### ISSUE THREAD

**Issue by songyuc** [pytorch/vision#1964]
## ❓ The simplest way to use checkpoint to maximize the GPU memory usage

Hi, guys,
I am learning about how to use the checkpoint to optimize the GPU memory usage, and I see there is a example in [densenet.py](https://github.com/pytorch/vision/blob/216035315185edec747dca8879d7197e7fb22c7d/torchvision/models/densenet.py#L53) as
```python
    @torch.jit.unused  # noqa: T484
    def call_checkpoint_bottleneck(self, input):
        # type: (List[Tensor]) -> Tensor
        def closure(*inputs):
            return self.bn_function(*inputs)

        return cp.checkpoint(closure, input)
```
Firstly, I think using checkpoint to maximize the GPU memory usage **only apply to activation modules, such as ReLU**. So, why not just create an new Module like:
```python
class cp_ReLU(nn.Module):
    def __init__(self, inplace) -> None:
        super(cp_ReLU, self).__init__()
        
        relu = nn.ReLU(inplace = inplace)
    
    def forward(self, x):
        y=cp.checkpoint(relu, x)
        return y
```
And use cp_ReLU instead of original ReLU in all the places where a ReLU is need as:
```python
if self.memory_efficient:
    self.add_module('relu2', nn.ReLU(inplace=True)),
else 
    self.add_module('relu2', cp_ReLU(inplace=True)),
```
I think this kind of implementation will make best use of the checkpoint.
Am I right?
Or would checkpoint also have effect to other kinds of modules like, Conv2d or BatchNorm2d?

Your suggestion and answer will be appreciated!

**Comment by fmassa** [comment]
Hi,

We use GitHub to track features / bug requests.
Questions like this are better suited for the discussion forums http://discuss.pytorch.org/, could you post your question there?

Thanks!

**Comment by songyuc** [comment]
Hi, @fmassa  Francisco,
in fact, I don't quite understand "Conv2d has intermediate activations", as far as I know, there is no activation function in Conv2d, which is a linear mapping operation.

**Comment by fmassa** [comment]
What I meant by activation is that it has an output, and the output is stored in memory for backpropagation, and thus it is affected by `checkpoint`
